docs/docstring_guidelines.md

```diff
@@ -0,0 +1,200 @@
+# Docstring guidelines
+
+Guidelines for writing docstrings for temporian's public API.
+
+Note that these guidelines are useful for internal docstrings too (a.k.a. the
+ones intended as documentation for other developers of the library rather than
+its users) but we do not enforce it on those. Internal docstrings can be leaner.
+
+## Objective
+
+The only purpose of reference material in general is to describe, as succinctly
+as possible, and in a consistent and orderly way. Good technical reference is
+essential to provide users with the confidence to do their work.
+
+Users hardly _read_ reference material; they _consult_ it. A docstring's
+objective is not to delight its readers with vocabulary and style.
+
+## General guidelines
+
+Docstrings should be:
+
+- Austere.
+- Accurate.
+- To the point.
+- Unambiguous.
+- Reliable.
+- Consistent (in language, structure, terminology and tone).
+
+Docstrings should do nothing but describe.
+
+Docstrings should not attempt to show how to perform tasks, but may include a
+succint example of the correct way to use what its documenting.
+
+## Style
+
+Temporian uses [Google's docstring style](https://google.github.io/styleguide/pyguide.html#38-comments-and-docstrings).
+
+Note that the descriptions below are a distilled and opinionated version of that
+document.
+
+## Concrete style guidelines
+
+### General
+
+Use three-double-quote format `"""`.
+
+A docstring should be composed of:
+
+1. A summary line (one physical line not exceeding 80 chars) terminated by a
+   period.
+2. A blank line.
+3. When writing more (recommended) the rest of the docstring.
+
+Every file should contain license boilerplate.
+
+Docstrings that do not provide any new information should be avoided:
+
+```
+def lag_operator_test(self):
+    """Test for lag operator."""
+```
+
+### Modules
+
+A module's docstring should describe the contents and usage of the module.
+
+Note that in Python every file is a module.
+
+In temporian we do not (yet) consider module docstrings to be mandatory.
+
+Example:
+
+```
+    """A one-line summary of the module, terminated by a period.
+
+    Leave one blank line.  The rest of this docstring should contain an
+    overall description of the module.  Optionally, it may also contain a brief
+    description of exported classes and functions and/or usage examples.
+
+    Typical usage example:
+
+    foo = ClassFoo()
+    bar = foo.FunctionBar()
+    """
```

Comment:
Do we allow md formatting in docstring? For example!:

"""
A one-line summary of the module, terminated by a period.

  Leave one blank line.  The rest of this docstring should contain an
  overall description of the module.  Optionally, it may also contain a brief
  description of exported classes and functions and/or usage examples.
  Typical usage example:

```python
  foo = ClassFoo()
  bar = foo.FunctionBar()
```

"""

Reply:
not sure if mkdocs supports it - cc @MNLucas